server/src/main/java/org/opensearch/action/admin/cluster/remotestore/stats/RemoteStoreStats.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.admin.cluster.remotestore.stats;
+
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.io.stream.Writeable;
+import org.opensearch.core.xcontent.ToXContentFragment;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.index.remote.RemoteRefreshSegmentTracker;
+
+import java.io.IOException;
+
+/**
+ * Encapsulates all remote store stats
+ *
+ * @opensearch.internal
+ */
+public class RemoteStoreStats implements Writeable, ToXContentFragment {
+
+    private final RemoteRefreshSegmentTracker.Stats remoteSegmentUploadShardStats;
+
+    public RemoteStoreStats(RemoteRefreshSegmentTracker.Stats remoteSegmentUploadShardStats) {
+        this.remoteSegmentUploadShardStats = remoteSegmentUploadShardStats;
+    }
+
+    public RemoteStoreStats(StreamInput in) throws IOException {
+        remoteSegmentUploadShardStats = in.readOptionalWriteable(RemoteRefreshSegmentTracker.Stats::new);
+    }
+
+    public RemoteRefreshSegmentTracker.Stats getStats() {
+        return remoteSegmentUploadShardStats;
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject()
+            .field("shardId", remoteSegmentUploadShardStats.shardId)
+            .field("latest_upload_files_count", remoteSegmentUploadShardStats.latestUploadFilesCount)
+            .field("latest_local_files_count", remoteSegmentUploadShardStats.latestLocalFilesCount)
+            .field("local_refresh_time", remoteSegmentUploadShardStats.localRefreshTime)
+            .field("local_refresh_seq_no", remoteSegmentUploadShardStats.localRefreshSeqNo)
+            .field("remote_refresh_time", remoteSegmentUploadShardStats.remoteRefreshTime)
+            .field("remote_refresh_seqno", remoteSegmentUploadShardStats.remoteRefreshSeqNo)
+            .field("bytes_lag", remoteSegmentUploadShardStats.bytesLag)
+            .field("inflight_upload_bytes", remoteSegmentUploadShardStats.inflightUploadBytes)
+            .field("inflight_uploads", remoteSegmentUploadShardStats.inflightUploads)
+            .field("rejection_count", remoteSegmentUploadShardStats.rejectionCount)
+            .field("consecutive_failure_count", remoteSegmentUploadShardStats.consecutiveFailuresCount);
+        builder.startObject("last_upload_time").field("started", -1).field("succeeded", -1).field("failed", -1);
```

Comment:
Why are these values set as `-1`?

Reply:
these are the stats which we are not added/integrated yet. When we add these in future, stats api can be updated to reflect the relevant value